test/models/project_test.rb

```diff
@@ -406,6 +406,33 @@ def clone_repository(project)
     end
   end
 
+  describe "#build_docker_image_for_branch?" do
+    context 'when the docker release branch is set' do
+      before do
+        project.docker_release_branch = 'master'
+        project.release_branch = 'master-of-puppets'
+      end
+      it 'returns false for the wrong branch' do
+        refute project.build_docker_image_for_branch?('master-of-puppets')
+      end
+
+      it 'returns true for the right branch' do
+        assert project.build_docker_image_for_branch?('master')
+      end
+    end
+
+    context 'when the docker release branch is not set' do
+      before do
+        project.docker_release_branch = nil
+        project.release_branch = 'master-of-puppets'
+      end
+      it 'returns false' do
+        refute project.build_docker_image_for_branch?(nil)
+        refute project.build_docker_image_for_branch?('master-of-puppets')
+      end
+    end
+  end
```

Comment:
This might be simpler:

```Ruby
describe "#build_docker_image_for_branch?" do
    before do
      project.docker_release_branch = 'master'
    end

    it 'returns true for the right branch' do
      assert project.build_docker_image_for_branch?('master')
    end

    it 'returns false for the wrong branch' do
      refute project.build_docker_image_for_branch?('other')
    end

    it 'returns false when not set' do
      project.docker_release_branch = nil
      refute project.build_docker_image_for_branch?(nil)
      refute project.build_docker_image_for_branch?('other')
    end
  end
```